### PR TITLE
Added component for content buffering time

### PR DESCRIPTION
**Pull request by VladyslavAnokhin**
## Changes

* Added component for content buffering time
* Remove duplicates in other components which store buffering time for mp4 ad and VRM processing.

[JIRA Ticket](https://jira.ouroath.com/browse/VSDK-4462)

@VerizonAdPlatforms/mobile-sdk-developers: Please review.

### CODE REVIEW

File: PlayerCore/components/ContentBufferingTime.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
+//  Copyright 2019, Oath Inc.
+//  Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
+
+import Foundation
+
+public enum ContentBufferingTime {
```
Comment (VladyslavAnokhin) [inline PlayerCore/components/ContentBufferingTime.swift]: We have the same component for mp4 ad buffering time. Maybe we can merge them?

Comment (AndriiMoskvin) [reply PlayerCore/components/ContentBufferingTime.swift]: If this component is agnostic to whether it is an ad or content - than it makes sense to reuse - and just have separation on toplevel state.

Comment (VladyslavAnokhin) [reply PlayerCore/components/ContentBufferingTime.swift]: Agree, will refactor this.

**Comment by AndriiMoskvin** [comment]
@VladyslavAnokhin is it RFR?

**Comment by VladyslavAnokhin** [comment]
@AndriiMoskvin Yes

File: PlayerCore/phantom types/BufferingStatus.swift [hunk 2]
```diff
@@ -0,0 +1,10 @@
+//  Copyright 2019, Oath Inc.
+//  Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
+
+import Foundation
+
+public enum BufferingStatus {
+    case inProgress(startAt: Date)
+    case finished(startAt: Date, finishAt: Date)
+    case empty
```
Comment (AndriiMoskvin) [inline PlayerCore/phantom types/BufferingStatus.swift]: What means `empty`?

Comment (VladyslavAnokhin) [reply PlayerCore/phantom types/BufferingStatus.swift]: There is nothing to buffer.

Comment (AndriiMoskvin) [reply PlayerCore/phantom types/BufferingStatus.swift]: `none` or `nothing`?

**Pull request merged**